websocket_push_session.go

```diff
@@ -229,8 +256,32 @@ func (s *wsPushSession) listen() {
 
 			pushConfig := elemental.NewPushConfig()
 			if err := elemental.Decode(s.encodingRead, data, pushConfig); err != nil {
-				s.close(websocket.CloseUnsupportedData)
-				return
+				if s.handlesErrorEvents() {
+					s.setErrorState(true)
+					msgpack, json, err := prepareEventData(elemental.NewErrorEvent(elemental.Error{
+						Description: fmt.Sprintf("could not decode message into %T: %s", pushConfig, err),
+						Subject:     "bahamut",
+						Title:       "Bad request",
+					}, s.encodingWrite))
```

Comment:
@primalmotion This can be made more efficient as you already know the encoding the client expects. Right now, it is generating both the MSGPACK and JSON encoding of the event which is a bit unnecessary.

I did this because the error path is not hot and I wanted to re-use the `prepareEventData` helper.

Reply:
I guess that's good enough for now. we can optimize later if needed